libdrgn/program.c

```diff
@@ -534,6 +583,11 @@ drgn_program_set_core_dump(struct drgn_program *prog, const char *path)
 	if (prog->core_fd == -1)
 		return drgn_error_create_os(errno, path, "open");
 
+	if (has_kdump_signature(prog->core_fd, &err))
```

Comment:
As a side note, normal ELF vmcores don't include vmalloc or per CPU addresses in the program headers, so drgn currently can't read from them. It seems like libkdumpfile can, so perhaps as a followup change, we should also use libkdumpfile for normal ELF vmcores.